### PR TITLE
Extended logging

### PR DESCRIPTION
**Pull request by thorstenkramm**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -190,10 +206,12 @@ func (c *Client) connectionLoop(ctx context.Context) {
 		if !isPrimary {
 			go func() {
 				for {
+					switchbackTimer := time.NewTimer(c.configHolder.Client.ServerSwitchbackInterval)
+					defer switchbackTimer.Stop()
```
Comment (morestatic) [inline client/client.go:210]: we should move this to an explicit stop after the `ctx.Done`

File: server/client_status_check.go [hunk 2]
```diff
@@ -81,13 +81,15 @@ func (t *ClientsStatusCheckTask) PingClients(clientsToPing <-chan *clients.Clien
 		if !ok && err == nil && string(response) == "unknown request" {
 			t.log.Debugf("ping to %s [%s] succeeded in %s. client < 0.8.2", cl.Name, cl.ID, rtt)
 			cl.LastHeartbeatAt = &now
+			cl.DisconnectedAt = nil
```
Comment (morestatic) [inline server/client_status_check.go:84]: this should use `SetConnected`

File: server/client_status_check.go [hunk 3]
```diff
@@ -81,13 +81,15 @@ func (t *ClientsStatusCheckTask) PingClients(clientsToPing <-chan *clients.Clien
...
 			results <- true
 			continue
 		}
 		// Only an empty response confirms the ping
 		if ok && err == nil && len(response) == 0 {
 			t.log.Debugf("ping to %s [%s] succeeded in %s. client >= 0.8.2", cl.Name, cl.ID, rtt)
 			cl.LastHeartbeatAt = &now
+			cl.DisconnectedAt = nil
```
Comment (morestatic) [inline server/client_status_check.go:92]: this should use `SetConnected`

File: server/clients/client_service.go [hunk 4]
```diff
@@ -613,6 +613,7 @@ func (s *ClientServiceProvider) SetLastHeartbeat(clientID string, heartbeat time
 		return err
 	}
 	existing.LastHeartbeatAt = &heartbeat
+	existing.DisconnectedAt = nil
```
Comment (morestatic) [inline server/clients/client_service.go:616]: this should use `SetConnected`

File: client/client.go (hunk 1 above)
Comment (thorstenkramm) [reply client/client.go:210]: @morestatic Can you please do?

**Review by morestatic — APPROVED** [review]

**Pull request merged**